core/src/test/scala/unit/kafka/server/metadata/BrokerMetadataListenerTest.scala

```diff
@@ -166,15 +172,15 @@ class BrokerMetadataListenerTest {
       image = newImage
     }
 
-    override def publishedOffset: Long = -1
+    override def publishedEndOffset: Long = -1
   }
 
   private val FOO_ID = Uuid.fromString("jj1G9utnTuCegi_gpnRgYw")
   private val BAR_ID = Uuid.fromString("SzN5j0LvSEaRIJHrxfMAlg")
 
   private def generateManyRecords(listener: BrokerMetadataListener,
                                   endOffset: Long): Unit = {
```

Comment:
I think there is some inconsistency here. This field is called `endOffset` and it is passed to `mockBatchReader`. `mockBatchReader` calls this parameter `lastOffset`. Kafka tends to make "endOffsets" exclusive while "lastOffsets" are inclusive.

Looking at the implementation of `mockBatchReader`, `lastOffset` is treated as the inclusive last offset of the batch.

Reply:
Got it. The part that I don't understand is that we send that value (`endOffset`) to `mockBatchReader` in the first parameter (`lastOffset`). Looking at the implementation of `mockBatchReader`, it looks to me like that code assumes that the offset is inclusive.